### PR TITLE
Scalar function for url encoding and decoding

### PR DESCRIPTION
**Pull request by jasperjiaguo**
## Description
Add scalar functions to perform url encoding and decoding in queries.
## Upgrade Notes
Does this PR prevent a zero down-time upgrade? (Assume upgrade order: Controller, Broker, Server, Minion)
No

### CODE REVIEW

File: pinot-broker/src/test/java/org/apache/pinot/broker/requesthandler/LiteralOnlyBrokerRequestTest.java [hunk 1]
```diff
@@ -94,6 +94,15 @@ public void testLiteralOnlyTransformBrokerRequestFromSQL() {
         .compileToPinotQuery("SELECT ago('PT1H'), fromDateTime('2020-01-01 UTC', 'yyyy-MM-dd z') FROM myTable")));
     Assert.assertFalse(BaseBrokerRequestHandler
         .isLiteralOnlyQuery(CalciteSqlParser.compileToPinotQuery("SELECT count(*) from foo where bar > ago('PT1H')")));
+    Assert.assertTrue(BaseBrokerRequestHandler.isLiteralOnlyQuery(CalciteSqlParser
```
Comment (siddharthteotia) [inline pinot-broker/src/test/java/org/apache/pinot/broker/requesthandler/LiteralOnlyBrokerRequestTest.java:97]: Should we not use full URLs as inputs to encodeUrl / decodeUrl functions in tests ? From user perspective, they will use the full URL right ?

`SELECT encodeUrl(<myURL>) FROM FOO .....`

File: pinot-common/src/main/java/org/apache/pinot/common/function/scalar/StringFunctions.java [hunk 2]
```diff
@@ -443,4 +446,28 @@ public static boolean contains(String input, String substring) {
...
+  /**
+   *
+   * @param input plaintext string
+   * @return url encoded string
+   * @throws UnsupportedEncodingException
+   */
+  @ScalarFunction
+  public static String encodeUrl(String input)
+      throws UnsupportedEncodingException {
+      return URLEncoder.encode(input, StandardCharsets.UTF_8.toString());
```
Comment (Jackie-Jiang) [inline pinot-common/src/main/java/org/apache/pinot/common/function/scalar/StringFunctions.java:459]:
```suggestion
      return URLEncoder.encode(input, StandardCharsets.UTF_8);
```

File: pinot-common/src/main/java/org/apache/pinot/common/function/scalar/StringFunctions.java [hunk 3]
```diff
@@ -443,4 +446,28 @@ public static boolean contains(String input, String substring) {
...
+  /**
+   *
+   * @param input url encoded string
+   * @return plaintext string
+   * @throws UnsupportedEncodingException
+   */
+  @ScalarFunction
+  public static String decodeUrl(String input)
+      throws UnsupportedEncodingException {
+    return URLDecoder.decode(input, StandardCharsets.UTF_8.toString());
```
Comment (Jackie-Jiang) [inline pinot-common/src/main/java/org/apache/pinot/common/function/scalar/StringFunctions.java:471]:
```suggestion
    return URLDecoder.decode(input);
```

**Review by Jackie-Jiang — APPROVED** [review]
LGTM otherwise

File: pinot-common/src/main/java/org/apache/pinot/common/function/scalar/StringFunctions.java (hunk 2 above)
Comment (jasperjiaguo) [reply pinot-common/src/main/java/org/apache/pinot/common/function/scalar/StringFunctions.java:459]: The java8 compilation will fail if we don't add `.toString()` 
https://github.com/apache/pinot/runs/5632328340?check_suite_focus=true

File: pinot-common/src/main/java/org/apache/pinot/common/function/scalar/StringFunctions.java (hunk 3 above)
Comment (jasperjiaguo) [reply pinot-common/src/main/java/org/apache/pinot/common/function/scalar/StringFunctions.java:471]: same reason as above

File: pinot-broker/src/test/java/org/apache/pinot/broker/requesthandler/LiteralOnlyBrokerRequestTest.java (hunk 1 above)
Comment (siddharthteotia) [reply pinot-broker/src/test/java/org/apache/pinot/broker/requesthandler/LiteralOnlyBrokerRequestTest.java:97]: @jasperjiaguo  - please add test cases with real URLs as well

Comment (Jackie-Jiang) [reply pinot-broker/src/test/java/org/apache/pinot/broker/requesthandler/LiteralOnlyBrokerRequestTest.java:97]: Sorry, I merged the PR too soon. @jasperjiaguo Can you please address Sidd's comments in a separate PR?

**Pull request merged**